### PR TITLE
CC-1251: Moving restore functions

### PR DESCRIPTION
**Pull request by smousa**
Originally, this was performed in the DFS--the idea here is to do this exclusively in the facade and either offer an interface to the dfs that calls these restores, or just leave this piece in the facade. (limiting the number of facade calls from the dfs)

### CODE REVIEW

**Comment by smousa** [comment]
retest this

File: facade/host_test.go [hunk 1]
```diff
@@ -93,6 +95,107 @@ func (s *FacadeTest) Test_HostCRUD(t *C) {
...
+	c.Assert(err, IsNil)
+	actual, err := s.Facade.GetHosts(s.CTX)
+	c.Assert(err, IsNil)
+	for i := range actual {
+		actual[i].DatabaseVersion = 0
+		actual[i].CreatedAt = time.Time{}
+		actual[i].UpdatedAt = time.Time{}
+	}
+	c.Assert(actual, DeepEquals, hosts1)
+	defer s.Facade.RemoveHost(s.CTX, "deadb11f")
```
Comment (tgjonestx) [inline facade/host_test.go]: Please move to line 123 just to be safe

File: facade/host_test.go [hunk 2]
```diff
@@ -93,6 +95,107 @@ func (s *FacadeTest) Test_HostCRUD(t *C) {
...
+				}, {
+					HostID:    "deadb11e",
+					IPAddress: "192.168.0.1",
+				},
+			},
+			CreatedAt: time.Time{},
+			UpdatedAt: time.Time{},
+		},
+	}
+	err = s.Facade.RestoreHosts(s.CTX, hosts2)
```
Comment (tgjonestx) [inline facade/host_test.go]: The result should be a non-nil `err`, right?

File: facade/host.go [hunk 3]
```diff
@@ -118,6 +118,42 @@ func (f *Facade) UpdateHost(ctx datastore.Context, entity *host.Host) error {
 	return err
 }
 
+// RestoreHosts restores a list of hosts, typically from a backup
+func (f *Facade) RestoreHosts(ctx datastore.Context, hosts []host.Host) error {
+	var exists bool
+	var err error
+
+	for _, host := range hosts {
+		host.DatabaseVersion = 0
```
Comment (tgjonestx) [inline facade/host.go]: Seems like it makes more sense to move this before line 144 so we're only changing properties of this host object if we know we're about to restore it.

File: facade/pool_test.go [hunk 4]
```diff
@@ -140,6 +140,79 @@ func (ft *FacadeTest) Test_RemoveResourcePool(t *C) {
...
+	c.Assert(err, IsNil)
+	actual, err := ft.Facade.GetResourcePools(ft.CTX)
+	c.Assert(err, IsNil)
+	for i := range actual {
+		actual[i].DatabaseVersion = 0
+		actual[i].CreatedAt = time.Time{}
+		actual[i].UpdatedAt = time.Time{}
+	}
+	c.Assert(actual, DeepEquals, pools1)
+	defer ft.Facade.RemoveResourcePool(ft.CTX, "testpool-1")
```
Comment (tgjonestx) [inline facade/pool_test.go]: Please move to line 161 just to be safe

File: facade/pool_test.go [hunk 5]
```diff
@@ -140,6 +140,79 @@ func (ft *FacadeTest) Test_RemoveResourcePool(t *C) {
...
+	c.Assert(err, IsNil)
+	actual, err = ft.Facade.GetResourcePools(ft.CTX)
+	c.Assert(err, IsNil)
+	for i := range actual {
+		actual[i].DatabaseVersion = 0
+		actual[i].CreatedAt = time.Time{}
+		actual[i].UpdatedAt = time.Time{}
+	}
+	c.Assert(actual, DeepEquals, pools2)
+	defer ft.Facade.RemoveResourcePool(ft.CTX, "testpool-2")
```
Comment (tgjonestx) [inline facade/pool_test.go]: Please move to line 204 just to be safe

File: facade/host_test.go (hunk 2 above)
Comment (smousa) [inline facade/host_test.go]: fixed this below

File: facade/host.go (hunk 3 above)
Comment (smousa) [inline facade/host.go]: The objects in this array are pass by value, so it wouldn't matter anyway

**Comment by tgjonestx** [comment]
:+1: Changes look good; just waiting for a green build

**Pull request merged**